src/clients/UBAClient/UBAClientBase.ts

```diff
@@ -246,78 +269,37 @@ export class BaseUBAClient extends BaseAbstractClient {
    * @param overrides The overrides to use for the LP fee calculation
    * @returns The system fee for the given token on the given chainId at the given block number
    */
-  public computeSystemFee(
+  public async computeSystemFee(
     depositChainId: number,
     destinationChainId: number,
     tokenSymbol: string,
+    hubPoolClient: HubPoolClient,
+    spokePoolClients: SpokePoolClients,
     amount: BigNumber,
     hubPoolBlockNumber: number,
     overrides?: UBALPFeeOverride
-  ): SystemFeeResult {
-    const [lpFee, { balancingFee: depositBalancingFee }] = [
-      this.computeLpFee(amount, depositChainId, destinationChainId, tokenSymbol, undefined, overrides),
-      this.computeBalancingFee(tokenSymbol, amount, hubPoolBlockNumber, depositChainId, UBAActionType.Deposit),
-    ];
+  ): Promise<SystemFeeResult> {
+    const lpFee = await this.computeLpFee(
+      hubPoolBlockNumber,
+      amount,
+      depositChainId,
+      destinationChainId,
+      hubPoolClient.chainId,
+      tokenSymbol,
+      hubPoolClient,
+      spokePoolClients,
+      overrides
+    );
+    const { balancingFee: depositBalancingFee } = this.computeBalancingFee(
+      tokenSymbol,
+      amount,
+      hubPoolBlockNumber,
+      depositChainId,
+      UBAActionType.Deposit
+    );
     return { lpFee, depositBalancingFee, systemFee: lpFee.add(depositBalancingFee) };
   }
 
-  /**
-   * Compute the entire UBA fee in the context of the UBA system for a given amount. The UBA fee is comprised of 7 return variables.
-   * @param depositChain The chainId of the deposit
-   * @param refundChain The chainId of the refund
-   * @param tokenSymbol The token to get the relayer fee for
-   * @param hubPoolTokenAddress The token address of the token on the hub pool chain
-   * @param amount The amount to get the relayer fee for
-   * @param hubPoolBlockNumber The block number to get the relayer fee for
-   * @param overrides The overrides to use for the LP fee calculation
-   * @returns The UBA fee for the given token on the given chainId at the given block number
-   */
-  public getUBAFee(
-    depositChain: number,
-    refundChain: number,
-    tokenSymbol: string,
-    amount: BigNumber,
-    hubPoolBlockNumber: number,
-    overrides?: UBALPFeeOverride
-  ): { relayerBalancingFee: BigNumber } & SystemFeeResult {
-    const [relayerFee, systemFee] = [
-      this.computeBalancingFee(tokenSymbol, amount, hubPoolBlockNumber, refundChain, UBAActionType.Refund),
-      this.computeSystemFee(depositChain, refundChain, tokenSymbol, amount, hubPoolBlockNumber, overrides),
-    ];
-    return {
-      relayerBalancingFee: relayerFee.balancingFee,
-      ...systemFee,
-    };
-  }
```

Comment:
Do we no longer need the Relay Balancing Fee?

Reply:
The Relayer doesn't use this. The relayer calls `computeSystemFee` and sets `realizedLpFee` using that value.

The Vercel API will call both `computeSystemFee` and `computeBalancingFee` to quote the `realizedLpFee` and the `relayerFeePct` respectively. Perhaps @dohaki might find this useful but I think the extra function adds more burden on maintainers and I found it difficult keeping this function up to date with all the internal changes I was making to `computeSystemFee` and `computeBalancingFee` so I've removed it from now.

Happy to add it once we have those internal functions more stable and finalized

Reply:
SGTM!